src/main/java/frc/robot/commands/TurnToHeadingWithDriving.java

```diff
@@ -0,0 +1,71 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+
+import frc.robot.subsystems.DriveTrain;
+
+
+public class TurnToHeadingWithDriving extends Command {
+
+  private final DriveTrain m_driveTrain;
+  private PIDController turnHeadingPID;
+  private Rotation2d m_wantedHeading;
+  private final DoubleSupplier m_translationXSupplier;
+  private final DoubleSupplier m_translationYSupplier;
+  private final double kp = 0.2; 
+  private final double ki = 0.0; 
+  private final double kd = 0.0; 
+
+  /**
+   * Creates a new autoAim.
+   */
+  public TurnToHeadingWithDriving(DriveTrain driveTrain, Supplier<Rotation2d> wantedHeading, DoubleSupplier translationXSupplier,
+      DoubleSupplier translationYSupplier) {
+    m_driveTrain = driveTrain;
+    m_wantedHeading = wantedHeading.get();
```

Comment:
No. You want to remember the Supplier here, and then call get() in initialize (and remember that value).

Reply:
Note that if you are using degrees internally, you can remember that value (in initialize) but probably should include "Degrees" in the variable name.

Reply:
i dont exactly understand what you mean but I think I've done this change